firestore-config/writes.test.js

```diff
@@ -73,22 +87,74 @@ test.each([
 test.each([
   ["anon", getAnonUser],
   ["stateless", getStatelessUser],
-  ["other state", getOutOfStateUser],
 ])("%s user cannot write", async (userType, getUserContext) => {
-  await testWriteToCollections(
+  await testWriteToCollectionsWithoutStateCode(
     UPDATE_COLLECTION_NAMES,
     getUserContext(testEnv).firestore(),
     assertFails
   );
 });
 
 test.each([
+  ["ND", getNDUser],
+  ["TN", getTNUser],
+])(
+  "%s user cannot write without stateCode",
+  async (userType, getUserContext) => {
+    await testWriteToCollectionsWithoutStateCode(
+      UPDATE_COLLECTION_NAMES,
+      getUserContext(testEnv).firestore(),
+      assertFails
+    );
+  }
+);
+
+test.each([["Recidiviz", getRecidivizUser]])(
+  "%s user can write without a stateCode",
+  async (userType, getUserContext) => {
+    await testWriteToCollectionsWithoutStateCode(
+      UPDATE_COLLECTION_NAMES,
+      getUserContext(testEnv).firestore(),
+      assertSucceeds
+    );
+  }
+);
+
+test.each([
+  ["Recidiviz", getRecidivizUser],
   ["TN", getTNUser],
+])("%s user can write TN data", async (userType, getUserContext) => {
+  await testWriteToCollectionsForState(
+    UPDATE_COLLECTION_NAMES,
+    getUserContext(testEnv).firestore(),
+    assertSucceeds,
+    "US_TN"
+  );
+});
+
+test.each([
   ["Recidiviz", getRecidivizUser],
-])("%s user can write", async (userType, getUserContext) => {
-  await testWriteToCollections(
+  ["ND", getNDUser],
+])("%s user can write ND data", async (userType, getUserContext) => {
+  await testWriteToCollectionsForState(
     UPDATE_COLLECTION_NAMES,
     getUserContext(testEnv).firestore(),
-    assertSucceeds
+    assertSucceeds,
+    "US_ND"
   );
 });
+
+test.each([
+  ["TN", getTNUser, "ND"],
+  ["ND", getNDUser, "TN"],
+])(
+  "%s user can write cross-state data",
```

Comment:
nit: user can not write cross-state data
```suggestion
  "%s user can not write cross-state data",
```